spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractMessageChannel.java

```diff
@@ -506,6 +504,33 @@ public boolean send(Message<?> message, long timeout) {
 		}
 	}
 
+	private Timer sendTimer(boolean sent) {
```

Comment:
Gary, I've taken a courage to refactor it like this:
```
private Timer sendTimer(boolean sent) {
		if (sent) {
			if (this.successTimer == null) {
				this.successTimer = buildSendTimer(true, "none");
			}
			return this.successTimer;
		}
		else {
			if (this.failureTimer == null) {
				this.failureTimer = buildSendTimer(false, "none");
			}
			return this.failureTimer;
		}
	}

	private Timer buildSendTimer(boolean success, String exception) {
		return Timer.builder(SEND_TIMER_NAME)
				.tag("type", "channel")
				.tag("name", getComponentName() == null ? "unknown" : getComponentName())
				.tag("result", success ? "success" : "failure")
				.tag("exception", exception)
				.description("Send processing time")
				.register(this.meterRegistry);
	}
```
So, now we even can use that `buildSendTimer()` for the exception part:
```
sample.stop(buildSendTimer(false, e.getClass().getSimpleName()));
```

Reply:
Cool; thanks.